common/views/components/IIIFImagePreview/IIIFImagePreview.js

```diff
@@ -0,0 +1,162 @@
+    };
+  }, []);
+
+  return (
+    <div className={`${classes} image-viewer__content image-viewer__content2`}>
+      <div className="image-viewer__controls flex flex-end flex--v-center">
+        <Control
+          type="light"
+          text="Zoom in"
+          id={`zoom-in-${id}`}
+          icon="zoomIn"
+          extraClasses={`${spacing({ s: 1 }, { margin: ['right'] })}`}
+          clickHandler={handleZoomIn}
+        />
+
+        <Control
+          type="light"
+          text="Zoom out"
+          id={`zoom-out-${id}`}
+          icon="zoomOut"
+          extraClasses={`${spacing({ s: 8 }, { margin: ['right'] })}`}
+          clickHandler={handleZoomOut}
+        />
+
+        <Control
+          type="light"
+          text="Close image viewer"
+          icon="cross"
+          extraClasses={`${spacing({ s: 2 }, { margin: ['right'] })}`}
+          clickHandler={() => {
+            handleViewerDisplay(null, 'Control');
+          }}
+        />
+      </div>
+
+      {viewerVisible && <ImageViewerImage id={id} infoUrl={infoUrl} />}
+    </div>
+  );
+};
+
+type IIIFImagePreviewProps = {|
+  iiifImageLocationUrl: string,
+|};
+const IIIFImagePreview = ({ iiifImageLocationUrl }: IIIFImagePreviewProps) => {
+  const [showViewer, setShowViewer] = useState(false);
+  const [jsEnabled, setJsEnabled] = useState(false);
+  const thumbnailUrl = iiifImageTemplate(iiifImageLocationUrl)({
+    size: ',400',
+  });
+  const largeSizeUrl = iiifImageTemplate(iiifImageLocationUrl)({
+    size: '1024,',
+  });
+
+  const handleViewerDisplay = (
+    event,
+    initiator: 'Button' | 'Control' | 'Image' | 'Keyboard'
+  ) => {
+    event && event.preventDefault();
+    trackEvent({
+      category: initiator,
+      action: `${showViewer ? 'closed' : 'opened'} ImageViewer`,
+      label: iiifImageLocationUrl,
+    });
+    setShowViewer(!showViewer);
+  };
+
+  useEffect(() => {
+    setJsEnabled(true);
+  }, []);
+
+  return (
+    <Fragment>
+      <img
+        className={classNames({
+          'cursor-zoom-in': jsEnabled,
+          [spacing({ s: 2 }, { margin: ['bottom'] })]: true,
+        })}
+        style={{ width: 'auto', display: 'block' }}
+        src={thumbnailUrl}
+        alt=""
+        onClick={() => {
+          handleViewerDisplay(null, 'Image');
+        }}
+      />
+      <Button
+        type="tertiary"
+        url={largeSizeUrl}
+        clickHandler={event => {
+          handleViewerDisplay(event, 'Button');
+        }}
+        text="View larger"
+      />
+      {showViewer && (
+        <ViewerContent
+          classes=""
+          viewerVisible={showViewer}
+          id={iiifImageLocationUrl}
+          handleViewerDisplay={handleViewerDisplay}
+          infoUrl={iiifImageLocationUrl}
+        />
+      )}
+    </Fragment>
+  );
+};
+
+export default IIIFImagePreview;
+// TODO put viewerContent in own file import here and in ImageViewer
+// TODO component / file names
```

Comment:
What structure do we envisage here?
We have split it up into quite a few pieces, and wonder if this is a good time to simplify?

Reply:
We also have loads of props passing, which makes me think it could be 1 component?